openmc/tallies.py

```diff
@@ -259,24 +264,69 @@ def with_summary(self):
 
     @property
     def sum(self):
+        if not self._statepoint:
```

Comment:
Actually, this may not be a good idea after all. Tally arithmetic would be difficult (and slow) without attributes dedicated to storing the arrays.